src/publisher_service/worker.rs

```diff
@@ -0,0 +1,258 @@
+                async move {
+                    process_queued_messages(&spawned_tasks_counter, &pg_pool, &relay_client).await
+                }
+            });
+        } else {
+            info!(
+                "Max number of workers {} reached, not spawning more",
+                MAX_WORKERS
+            );
+        }
+    }
+}
+
+/// Picking messages from the queue and processing them in a loop until
+/// there are no more messages to process
+#[instrument]
+async fn process_queued_messages(
+    spawned_tasks_counter: &Arc<AtomicUsize>,
+    pg_pool: &PgPool,
+    relay_client: &Arc<relay_client::http::Client>,
+) {
+    // Increasing the number of spawned tasks atomic counter
+    spawned_tasks_counter.fetch_add(1, Ordering::SeqCst);
+
+    // Querying for queued messages to be published in a loop until we are done
+    loop {
+        match pick_message_for_processing(pg_pool).await {
+            Ok(result) => {
+                if let Some(message_to_process) = result {
+                    info!("Got a message with id: {}", message_to_process.id);
+                    process_message(
+                        &message_to_process.project,
+                        &message_to_process.notification,
+                        &message_to_process.subscriber,
+                        pg_pool,
+                        relay_client.clone(),
+                    )
+                    .await
+                    .unwrap();
+
+                    update_message_processing_status(
+                        &message_to_process.id,
+                        NotificationStates::Published,
+                        pg_pool,
+                    )
+                    .await
+                    .unwrap();
+                } else {
+                    info!("No more messages to process, stopping the loop");
+                    break;
+                }
+            }
+            Err(_) => {
+                // TODO: process this error properly
+                // Decreasing the number of spawned tasks atomic counter
+                spawned_tasks_counter.fetch_sub(1, Ordering::SeqCst);
+                return;
+            }
+        }
+    }
+
+    // Decreasing the number of spawned tasks atomic counter
+    spawned_tasks_counter.fetch_sub(1, Ordering::SeqCst);
+}
+
+#[instrument]
+async fn pick_message_for_processing(
+    pg_pool: &PgPool,
+) -> Result<Option<ProcessingMessageItem>, sqlx::Error> {
+    // Getting the message to be published from the `subscriber_notification`,
+    // updating the status to the `processing`,
+    // and returning the message to be processed with the project id
+    let pick_message_query = "
+        WITH cte AS (
+            SELECT id
+            FROM subscriber_notification
+            WHERE status = 'queued'
+            LIMIT 1
+            FOR UPDATE SKIP LOCKED
+        )
+        UPDATE subscriber_notification
+        SET status = 'processing', updated_at = NOW()
```

Comment:
Looks like we are missing ability to reset processing state if publishing failed?